com.unity.render-pipelines.high-definition/Editor/Material/UIBlocks/MaterialUIBlock.cs

```diff
@@ -133,11 +134,23 @@ public enum ExpandableBit : uint
             User19 = 1 << 30,
         }
 
+        internal MaterialUIBlock()
+        {
```

Comment:
[question about c#] here i assume it is okay we initialize the values to their default value that dont need to be explicit?

Reply:
Yes, the default constructor on c# initializes:

- All numeric fields in the class to zero.
- All string and object fields to null.

So later on the method, `internal void Initialize(MaterialEditor materialEditor, MaterialProperty[] properties, MaterialUIBlockList parent)` everything will be handled.
If a `header` is not speficied and the `OnGUI` is not overrided, the dev will get the exception on `MaterialHeaderScope`, this along the other exception will handle both paths on any MaterialUIBlock

Reply:
If I get it well here you make the MaterialUIBlock non usable to user as you need to comply with the internal visibility to be able to create one of its child.

Is this something that we want? At that point, why MaterialUIBlock is even public? we can all works in package internal visibility.
Be sure you don't break up potential feature :)

------

And what you said about constructor is not 100% accurate so here is more explanation for Jenny.
The (default) constructor on c# initializes:
- All value type (numeric fields and struct) in the class to 0, 0f, default struct... if they don't have a provided initializer at the scope of the class.
- All object type (string, array and object fields) to null if they don't have a provided initializer at the scope of the class.
- Then once all is initialized, the content of the constructor (if any) runs and potentially add more value change.

Also, default constructor always exists and is public except specified differently.

Try this to be sure ;)
```
using UnityEngine;

#if UNITY_EDITOR
using UnityEditor;
#endif

public class tester : MonoBehaviour
{
    class TestClass
    {
        public int int_default;
        public int int_42 = 42;

        public class Klass
        {
            public int int_field;
            public Klass(int val) => int_field = val;
        }
        public Klass klass_default;
        public Klass klass_33 = new Klass(33);
    }
    
#if UNITY_EDITOR
    [MenuItem("CONTEXT/tester/Test")]
    static void Test()
    {
        TestClass tc = new TestClass();
        
        Debug.Log($"int_default={tc?.int_default}");                                                //expect 0
        Debug.Log($"int_42={tc?.int_42}");                                                          //expect 42
        Debug.Log($"klass_default={(tc?.klass_default?.int_field == null ? "NULL": "NOT NULL")}");  //expect null
        Debug.Log($"klass_33={tc?.klass_33?.int_field}");                                           //expect 33
    }
#endif
}
```